packages/wallet/api/src/sharingUnserializer.js

```diff
@@ -0,0 +1,81 @@
+// @ts-check
+import { Far, makeMarshal } from '@endo/marshal';
+
+/**
+ * Make a pair of unserializers that share low-privilege values.
+ *
+ * The "high" privilege unserializer can understand objects as slots, such as
+ * those from the wallet serializer. The "low" privilege unserializer can only
+ * understand strings as slots, such as those from the board serializer.
+ *
+ * This provides a way, for instance, to get identical brand references from
+ * both a public contract and a user's purses.
+ *
+ * @param {string} [lowProperty] - The key name to check on a high privilege
+ * slot object. If the string value at this key is equivalent to the value of
+ * another low privilege slot, the high privilege unserializer will return the
+ * same object for both slots.
+ */
+export const makeSharingUnserializer = lowProperty => {
+  assert(lowProperty, 'lowProperty must be specified for sharing unserializer');
+  const seen = {
+    // high: new WeakMap(),  // ISSUE: WeakMap conflicts with string keys
+    high: new Map(),
+    low: new Map(),
+  };
+
+  const slotToVal = {
+    /**
+     * @param {Record<string, any>} slot
+     * @param {string} iface
+     */
+    high: (slot, iface) => {
+      let obj;
+      let key;
+
+      if (lowProperty in slot) {
```

Comment:
this condition seems more like delegating to a fallback, breaking out of the primary logic of the function. Refactoring that way would make it more clear and also allow immutable bindings.

```js
if (lowProperty in slot) {
  return slotToVal.low(slot[lowProperty]);
}

const key = JSON.stringify(slot);
…
const obj = Far(iface, {});
```